src/main/java/com/bigchaindb/builders/BigchainDbTransactionBuilder.java

```diff
@@ -326,8 +326,10 @@ public IBuild build(EdDSAPublicKey publicKey) {
                 this.transaction.addInput(input);
             }
 
-            if (this.transaction.getOperation() == null) {
+            if (this.operation == null) {
                 this.transaction.setOperation("CREATE");
+            } else {
+                this.transaction.setOperation(this.operation.name());
```

Comment:
fix for #2, reported by @NikitaLos

Reply:
fair enough I'll try to beef this up with a better check, perhaps with an `IllegalArgumentException` if it's wrong

Reply:
on second thought, are you sure this is needed, @innoprenuer? it's typed as an `Operations` enum instance, so it probably can't easily be anything _but_ `CREATE` or `TRANSFER`.

Reply:
it will work but will not help with readability. But, in case on CREATE transaction just checking null won't do because we want users to explicitly mention what type of Operation they want to perform (so that it is same as TRANSFER transaction)

```
   if (this.operation == Operations.CREATE) {
            	this.transaction.setOperation("CREATE");
            }
   else if(this.operation == Operations.TRANSFER){
        	this.transaction.setOperation("TRANSFER");	
        }
```